core/src/executor/fetch.rs

```diff
@@ -166,33 +169,48 @@ pub async fn fetch_relation_columns(
             fetch_columns(storage, table_name).await
         }
         TableFactor::Derived {
-            subquery:
-                Query {
-                    body: SetExpr::Select(statement),
+            subquery: Query { body, .. },
+            alias: TableAlias { columns, name },
+        } => match body {
+            SetExpr::Select(statement) => {
+                let Select {
+                    from:
+                        TableWithJoins {
+                            relation, joins, ..
+                        },
+                    projection,
                     ..
-                },
-            alias: _,
-        } => {
-            let Select {
-                from: TableWithJoins {
-                    relation, joins, ..
-                },
-                projection,
-                ..
-            } = statement.as_ref();
+                } = statement.as_ref();
+
+                let columns = fetch_relation_columns(storage, relation).await?;
+                let join_columns = fetch_join_columns(joins, storage).await?;
+                let labels = get_labels(
+                    projection,
+                    get_alias(relation)?,
+                    &columns,
+                    Some(&join_columns),
+                )?;
+
+                Ok(labels)
+            }
+            SetExpr::Values(Values(values_list)) => {
+                let width = values_list[0].len();
+                let alias_len = columns.len();
+                let labels = (alias_len + 1..=width)
+                    .into_iter()
+                    .map(|i| format!("column{}", i));
+                let labels = match alias_len > width {
+                    true => {
+                        return Err(
+                            FetchError::TooManyColumnAliases(name.into(), width, alias_len).into(),
+                        )
+                    }
+                    false => columns.iter().cloned().chain(labels).collect::<Vec<_>>(),
+                };
```

Comment:
It looks that we can make make code a bit more simple by doing some rearrangement.
```rust
let total_len = values_list[0].len();
let alias_len = columns.len();

if alias_len > total_len {
    return Err(FetchError::TooManyColumnAliases(
        name.into(),
        total_len,
        alias_len,
    )
    .into());
}

let labels = (alias_len + 1..=total_len)
    .into_iter()
    .map(|i| format!("column{}", i));
let labels = columns.iter().cloned().chain(labels).collect::<Vec<_>>();
```